### PR TITLE
Improve Evaluations

### PR DESCRIPTION
**Pull request by soujanyanmbri**
> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Note**: _that go.mod and go.sum can only be modified for tested dependency updates or justified new features._

**Make sure that you've checked the boxes below before you submit PR:**

- [ ] My code includes positive and negative tests.
- [ ] I have an appropriate description with correct grammar.
- [ ] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v2sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).

Thanks for your PR :heart:

### CODE REVIEW

File: experimental/types/value_metadata.go [hunk 1]
```diff
@@ -45,154 +45,157 @@ const (
...
+var metadataStrings = map[string]DataMetadata{
+	"numeric":          ValueMetadataNumeric,
+	"boolean":          ValueMetadataBoolean,
+	"alphanumeric":     ValueMetadataAlphanumeric,
+	"ascii":            ValueMetadataAscii,
+	"base64":           ValueMetadataBase64,
+	"uri":              ValueMetadataURI,
+	"domain":           ValueMetadataDomain,
+	"unicode":          ValueMetadataUnicode,
+	"not_numeric":      NotValueMetadataNumeric,
```
Comment (piyushroshan) [inline experimental/types/value_metadata.go:63]: I was thinking if we can even remove the negated ones. So have one source of truth

Comment (soujanyanmbri) [reply experimental/types/value_metadata.go:63]: Done

File: experimental/types/value_metadata.go [hunk 2]
```diff
@@ -45,160 +45,168 @@ const (
...
-	return res
+	metadata[ValueMetadataUnicode] = EvaluationData{Result: false, Evaluated: true}
 }
 
 func (v *DataMetadataList) IsInScope(metadataTypes []DataMetadata) bool {
 	for _, metadataType := range metadataTypes {
-		if v.metadata[metadataType] {
-			return true
+		// Check if metadataType starts with not_
+		if metadataType >= NotValueMetadataAlphanumeric {
```
Comment (soujanyanmbri) [inline experimental/types/value_metadata.go:202]: Change logic to not depend on the order

File: experimental/types/value_metadata.go [hunk 3]
```diff
@@ -45,160 +45,168 @@ const (
 	NotValueMetadataUnicode
 )
 
+type EvaluationData struct {
+	Evaluated bool
+	Result    bool
+}
+
+// MetadataStrings provides a mapping of strings to metadata for quick lookup.
+var metadataStrings = map[string]DataMetadata{
```
Comment (soujanyanmbri) [inline experimental/types/value_metadata.go:54]: Add map here, we can do reverseMap to get not

**Pull request merged**